### PR TITLE
ceph: fix object store healthcheck yaml

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

Reverting some of
23f0ac7
to re-use `bucket` for the healthcheck configuration.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 1]
```diff
@@ -450,7 +450,7 @@ type ObjectStoreSpec struct {
 }
 
 type BucketHealthCheckSpec struct {
-	Bucket        HealthCheckSpec   `json:"rgw,omitempty"`
+	Bucket        HealthCheckSpec   `json:"bucket,omitempty"`
```
Comment (galexrt) [inline pkg/apis/ceph.rook.io/v1/types.go:453]: Should a breaking change entry be added to the pending release notes for this rename?

Comment (leseb) [reply pkg/apis/ceph.rook.io/v1/types.go:453]: This must be backported, so updating the pending release note won't help much, unfortunately. We will mention this once we release 1.4.2. How does that sound?

Comment (galexrt) [reply pkg/apis/ceph.rook.io/v1/types.go:453]: I see, yeah that should suffice.

File: cluster/examples/kubernetes/ceph/object.yaml [hunk 2]
```diff
@@ -86,8 +86,8 @@ spec:
     #name: zone-a
   # service endpoint healthcheck
   healthCheck:
-    endpoint:
-      enabled: true
+    bucket:
+      disabled: false
```
Comment (galexrt) [inline cluster/examples/kubernetes/ceph/object.yaml:90]: One last thing that came to my mind, is the Documentation also correct now?

Comment (leseb) [reply cluster/examples/kubernetes/ceph/object.yaml:90]: Yes!

**Pull request merged**